cmd/autobumper/main.go

```diff
@@ -1,7 +1,89 @@
-// copied from go/src/k8s.io/test-infra/experiment/autobumper/main.go
-// originally based on: 3c673e642 (HEAD -> master, origin/master, origin/HEAD) Merge pull request #13978 from nilo19/t-qini-update_k8s_and_aks_engine_version_for_periodic_tests
 package main
 
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/experiment/autobumper/bumper"
+	"k8s.io/test-infra/prow/config/secret"
+	"k8s.io/test-infra/prow/github"
+)
+
+const (
+	githubOrg  = "openshift"
+	githubRepo = "release"
+)
+
+var extraFiles = map[string]bool{
+	"hack/images.sh": true,
+}
+
+type options struct {
+	githubLogin string
+	githubToken string
+	gitName     string
+	gitEmail    string
+	targetDir   string
+}
+
+func parseOptions() options {
+	var o options
+	flag.StringVar(&o.githubLogin, "github-login", "", "The GitHub username to use.")
+	flag.StringVar(&o.githubToken, "github-token", "", "The path to the GitHub token file.")
+	flag.StringVar(&o.gitName, "git-name", "", "The name to use on the git commit. Requires --git-email. If not specified, uses the system default.")
+	flag.StringVar(&o.gitEmail, "git-email", "", "The email to use on the git commit. Requires --git-name. If not specified, uses the system default.")
+	flag.StringVar(&o.targetDir, "target-dir", "", "The directory containing the target repo.")
+	flag.Parse()
+	return o
+}
+
+func validateOptions(o options) error {
+	if o.githubLogin == "" {
+		return fmt.Errorf("--github-login is mandatory")
+	}
+	if o.githubToken == "" {
+		return fmt.Errorf("--github-token is mandatory")
+	}
+	if (o.gitEmail == "") != (o.gitName == "") {
+		return fmt.Errorf("--git-name and --git-email must be specified together")
+	}
+	if o.targetDir == "" {
+		return fmt.Errorf("--target-dir is mandatory")
+	}
+	return nil
+}
+
 func main() {
+	o := parseOptions()
+	if err := validateOptions(o); err != nil {
+		logrus.WithError(err).Fatal("Invalid arguments.")
+	}
+
+	sa := &secret.Agent{}
+	if err := sa.Start([]string{o.githubToken}); err != nil {
+		logrus.WithError(err).Fatal("Failed to start secrets agent")
+	}
+
+	gc := github.NewClient(sa.GetTokenGenerator(o.githubToken), sa.Censor, github.DefaultGraphQLEndpoint, github.DefaultAPIEndpoint)
+
+	logrus.Infof("Changing working directory to %s...", o.targetDir)
+	if err := os.Chdir(o.targetDir); err != nil {
+		logrus.WithError(err).Fatal("Failed to change to root dir")
+	}
+	images, err := bumper.UpdateReferences([]string{"cluster/ci/config/prow/", "ci-operator/", "hack/"}, extraFiles)
```

Comment:
If we are providing `hack/` entirely why do we need `extraFiles`?

Reply:
The scanning focuses on `*.yaml` files.